inc/c_util/constbuffer.h

```diff
@@ -65,6 +65,8 @@ MOCKABLE_INTERFACE(constbuffer,
 
     FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithMoveMemory, unsigned char*, source, uint32_t, size),
 
+    FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithMoveMemoryWithCustomFree, unsigned char*, source, uint32_t, size, CONSTBUFFER_CUSTOM_FREE_FUNC, customFreeFunc, void*, customFreeFuncContext),
```

Comment:
>CONSTBUFFER_CUSTOM_FREE_FUNC [](http://example.com/codeflow?start=123&length=28)

I don't think this does what the name of the API implies it does.

The API should get another pointer function that does "move". (CreateWith*Move*).

Otherwise, how does it know how to "move"? It needs to tell to the _current_ owner something along the lines of "I am stealing this memory from under you, and I promise I will call your own free at a later time".

The "steal" part is missing.

Reply:
Normally I would be waiting, but this is not a real code review, just something to get a preview of what the collective is thinking.